### PR TITLE
feat(minify shadowroots): add rollup-plugin-minify-html-literals

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

No ticket

### Description

Adds [rollup-plugin-minify-html-literals](https://www.npmjs.com/package/rollup-plugin-minify-html-literals) to the build pipeline. This minifies the HTML resulting from each component's `render()` function.

### Changelog

**Changed**

- Added rollup-plugin-minify-html-literals to rollup config

### CODE REVIEW

**Comment by andy-blum** [comment]
Local comparisons:

### BXAccordion & BXAccordionItem

|            | Before | After | % Reduction |
|------------|--------|-------|-------------|
| Nodes      |    102 |    61 |         40% |
| Comments   |     17 |    20 |        -18% |
| Whitespace |     34 |     1 |         97% |

### DDSButtonGroup & DDSButtonGroupItem

|            | Before | After | % Reduction |
|------------|--------|-------|-------------|
| Nodes      |     53 |    38 |         28% |
| Comments   |      6 |    12 |       -100% |
| Whitespace |     24 |     3 |         88% |

File: packages/carbon-web-components/tools/get-rollup-config.js [hunk 1]
```diff
@@ -129,6 +130,17 @@ function getRollupConfig({
...
+        failOnError: true,
+        options: {
+          minifyOptions: {
+            caseSensitive: true,
+            collapseInlineTagWhitespace: true,
+            collapseWhitespace: true,
+            removeComments: true,
+          },
+        },
+      }),
```
Comment (annawen1) [inline packages/carbon-web-components/tools/get-rollup-config.js:143]: There's a section below that is supposed to minify the html templates here: 
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/tools/get-rollup-config.js#L198-#L216

I believe kenny put a comment in saying it wasn't working, so maybe we can remove that redundant code and replace it with this plugin?

**Comment by andy-blum** [comment]
[Codesandbox demo](https://codesandbox.io/s/vibrant-shaw-y962t3?file=/index.html:0-895) of ibm.com/sustainability. Modules are autoloaded here

```
            Before  After
Nodes       9,830   5,879
Comments    1,800   2,347
Whitespace  4,816   488
```

**Comment by andy-blum** [comment]
Closing. Will re-open as new V1 & V2 PRs

**Pull request closed without merging**